sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobClient.java

```diff
@@ -109,6 +116,32 @@ public BlobOutputStream getBlobOutputStream(ParallelTransferOptions parallelTran
             metadata, tier, requestConditions);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void uploadWithResponse(InputStream data, long length, ParallelTransferOptions parallelTransferOptions,
+        BlobHttpHeaders headers, Map<String, String> metadata, AccessTier tier, BlobRequestConditions requestConditions,
+        Duration timeout, Context context) {
+        final ParallelTransferOptions validatedParallelTransferOptions =
+            ModelHelper.populateAndApplyDefaults(parallelTransferOptions);
+
+        // BlobOutputStream passes to buffered upload, which has logic for switching based on stream size
+        BlobOutputStream blobOutputStream = BlobOutputStream.blockBlobOutputStream(encryptedBlobAsyncClient,
```

Comment:
don't feel two strongly about this but could we make this logic protected in regular Blob so its not duplicated?

Reply:
Good call!